core/vec3.h

```diff
@@ -52,9 +52,14 @@ class Vec3 {
   float z() const;
   void setZ(float num);
 
+ 
 private:
   float x_, y_, z_;
 };
 
+static const Vec3 kVecZero(0, 0, 0);
```

Comment:
Please move those to the Vec3 class. Yes, what I showed does not compile, but you should be able to figure out how to make this work. Given you have read entire c++ book you should have also read about initializing static variables.